extensions/httpclient/src/test/groovy/com/stormpath/sdk/client/DirectoryIT.groovy

```
@@ -800,4 +810,126 @@ class DirectoryIT extends ClientIT {
+        directory = client.createDirectory(directory);
+        assertNotNull directory.href
+        deleteOnTeardown(directory)
+
+        def accountSchema = directory.getAccountSchema()
+        assertNotNull accountSchema
+        assertNotNull accountSchema.getHref()
+
+        def accountSchema1 = client.getResource(accountSchema.getHref(), Schema)
+
+        assertFalse(accountSchema.equals(accountSchema1))  //not expanded -> must be different
+    }
+
+    /**
+     * @since 1.1.0
+     */
+    @Test
+    void testGetDirectoryWithExpandedAccountSchema() {
+
+        Directory directory = client.instantiate(Directory)
+        directory.setName(uniquify("JSDK.DirectoryIT.testGetDirectoryWithExpandedAccountSchema"))
+        directory = client.createDirectory(directory);
+        assertNotNull directory.href
+        deleteOnTeardown(directory)
+
+        def dirOptions = Directories.options().withAccountSchema()
+
+        directory = client.getResource(directory.getHref(), Directory, dirOptions)
+
+        //accountSchema must be expanded
+        def accountSchema = directory.getAccountSchema()
+        assertNotNull accountSchema
+        assertNotNull accountSchema.getHref()
+
+        def accountSchema1 = client.getResource(accountSchema.getHref(), Schema)
+
+        assertEquals(accountSchema, accountSchema1)  //expanded -> must be equals
+    }
+
+    /**
+     * @since 1.2.0
+     */
+    @Test
+    public void testCreateAccountWithDefaultAccountSchema() {
+        def app = createTempApp()
+        //Directory directory = app.getDefaultAccountStore()
+        def username = uniquify('Stormpath-SDK-Test-App-Acct1')
+        def account = client.instantiate(Account)
+                .setUsername(username)
+                .setPassword("Changeme1!")
+                .setEmail(username + "@nowhere.com")
+        account = app.createAccount(Accounts.newCreateRequestFor(account).setRegistrationWorkflowEnabled(false).build())
```

This test should try to create an account a few times and all should fail because of missing fields required by the default schema. One must finally succeed because it matches the fields required by default.